src/nvim/ex_cmds.c

```diff
@@ -321,12 +321,10 @@ static int sort_compare(const void *s1, const void *s2)
     /* We need to copy one line into "sortbuf1", because there is no
      * guarantee that the first pointer becomes invalid when obtaining the
      * second one. */
-    STRNCPY(sortbuf1, ml_get(l1.lnum) + l1.start_col_nr,
+    xstrlcpy(sortbuf1, (char *)ml_get(l1.lnum) + l1.start_col_nr,
```

Comment:
> The macro, STRLCPY would obviate the case.

@splinterofchaos it's actually a better idea not to use these macros and make the casting explicit. At least for new `char_u` related macros.